app/src/main/java/ru/firstSet/kotlinOne/utils/EventCalendar.kt

```diff
@@ -1,49 +1,62 @@
 package ru.firstSet.kotlinOne.utils
 
-import android.app.Application
-import android.content.ContentValues
-import android.content.res.Resources
-import android.net.Uri
+import android.app.DatePickerDialog
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
 import android.provider.CalendarContract
-import androidx.core.app.NotificationManagerCompat
+import android.widget.DatePicker
+import androidx.core.content.ContextCompat.startActivity
 import androidx.core.net.toUri
 import ru.firstSet.kotlinOne.R
 import ru.firstSet.kotlinOne.data.Movie
 import java.util.*
 
-class EventCalendar(val year: Int, val month: Int, val day: Int, val movie: Movie) {
-    var eventsUri = "content://calendar/movies/${movie.id}".toUri()
+class EventCalendar(val context: Context, val savedInstanceState: Bundle?, val movie: Movie) {
+    var yearPlan: Int = 2021
+    var monthPlan: Int = 2
+    var dayPlan: Int = 15
 
-    val startMillis: Long = Calendar.getInstance().run {
-        set(year, month, day, 19, 30)
-        timeInMillis
-    }
-    val endMillis: Long = Calendar.getInstance().run {
-        set(year, month, day, 22, 30)
-        timeInMillis
+    val dateSetListener = object : DatePickerDialog.OnDateSetListener {
+        override fun onDateSet(
+            view: DatePicker, year: Int, monthOfYear: Int,
+            dayOfMonth: Int
+        ) {
+            yearPlan = year
+            monthPlan = monthOfYear
+            dayPlan = dayOfMonth
+            createEventCalendar()
+        }
     }
 
-    fun setEvent() {
-        val values = ContentValues().apply {
-            put(CalendarContract.Events.DTSTART, startMillis)
-            put(CalendarContract.Events.DTEND, endMillis)
-            put(
-                CalendarContract.Events.TITLE,
-                Resources.getSystem().getString(R.string.calendar_contract_events_title)
-            )
-            put(
-                CalendarContract.Events.DESCRIPTION,movie.title
-            )
-            put(CalendarContract.Events.CALENDAR_ID, calID)
-            put(CalendarContract.Events.ALL_DAY, 1)
-            put(CalendarContract.Events.HAS_ALARM, 1)
+    fun showPickerDialog() {
+        val datePickerDialog = context.let {
+            DatePickerDialog(
+                it, dateSetListener, yearPlan, monthPlan, dayPlan
+            ).show()
         }
-        var url: Uri? = Application().contentResolver.insert(eventsUri, values)
     }
 
-    companion object{
-    val calID: Long = 1600
-
-}
-
+    private fun createEventCalendar() {
+        val eventsUri = "content://com.android.calendar/events".toUri()
+        val startMillis: Long = Calendar.getInstance().run {
+            set(yearPlan, monthPlan, dayPlan, 19, 30)
```

Comment:
@kamar19 Why there is also hard code? Looks like we need to get the values from peeker